idaes/surrogate/pysmo/tests/test_sampling.py

```diff
@@ -151,34 +151,39 @@ class TestSamplingMethods:
     @pytest.mark.parametrize("array_type", [np.array])
     def test_nearest_neighbour_01(self, array_type):
         input_array = array_type(self.test_data_3d)
+        self.x_data = np.zeros((2, input_array.shape[1] - 1))
```

Comment:
This is not specific to this PR, but "using" a pytest `TestXYZ` class as a normal Python object, i.e. setting and accessing class or instance attributes using `self.x_data` here, is generally discouraged as it can cause warnings and/or unexpected behavior.

The recommended way is to define a pytest fixture that's specific to the class: this is done by defining a normal method and decorating it with `@pytest.fixture`.